Refactor using testify and upgrade go v1.22.7

nitram509 opened the pull request:
Pavlo, as we wrote in the other issue about p12 support, 
I thought it's best to split the tasks. 
This is the full refactoring, so that `testify` lib is used in all tests.

Also, I did upgrade to Go 1.22.7 and this led to updates to golinter as well.
About the linter ... I spent quite some time fixing linter issues.
Therefore, you see more changes than necessary, because with Go 1.22, the linter went more "aggressive".

Any feedback is welcome.
Also, it would be great to get the 'hacktoberfest-approved' label ;)

nitram509 committed:
switch to Go v1.22.7

nitram509 committed:
refactor using stretchr/testify

nitram509 committed:
fix linter errors

nitram509 committed:
fix linter errors

nitram509 committed:
upgrade linter in GH action

pavlo-v-chernykh commented:
I’ve added the `hacktoberfest-accepted` label required by the participation rules. However, I couldn’t find any mention of the `hacktoberfest-approved` label on [Hacktoberfest Participation](https://hacktoberfest.com/participation). Please let me know what it is used for.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -343,7 +321,7 @@ func TestReadString(t *testing.T) {
 		str := "some string to read"
 		buf := make([]byte, 2)
-		binary.BigEndian.PutUint16(buf, uint16(len(str)))
+		binary.BigEndian.PutUint16(buf, uint16(len(str))) //nolint:all
```
Same here about nolint

pavlo-v-chernykh commented on encoder.go:
```diff
@@ -95,7 +95,7 @@ func (e encoder) writePrivateKeyEntry(alias string, pke PrivateKeyEntry) error {
 	}
 
-	if err := e.writeUint64(uint64(pke.CreationTime.UnixMilli())); err != nil {
+	if err := e.writeUint64(uint64(pke.CreationTime.UnixMilli())); err != nil { //nolint:all
```
Same here about nolint

pavlo-v-chernykh commented on encoder.go:
```diff
@@ -140,7 +140,7 @@ func (e encoder) writeTrustedCertificateEntry(alias string, tce TrustedCertifica
 	}
 
-	if err := e.writeUint64(uint64(tce.CreationTime.UnixMilli())); err != nil {
+	if err := e.writeUint64(uint64(tce.CreationTime.UnixMilli())); err != nil { //nolint:all
```
Same here about nolint

pavlo-v-chernykh commented on keystore.go:
```diff
@@ -117,7 +117,7 @@ func (ks KeyStore) Store(w io.Writer, password []byte) error {
 	}
 
-	if err := e.writeUint32(uint32(len(ks.m))); err != nil {
+	if err := e.writeUint32(uint32(len(ks.m))); err != nil { //nolint:all
```
Same here about nolint

pavlo-v-chernykh commented on keystore.go:
```diff
@@ -192,7 +192,7 @@ func (ks KeyStore) Load(r io.Reader, password []byte) error {
 	computedDigest := d.h.Sum(nil)
 
-	actualDigest, err := d.readBytes(uint32(d.h.Size()))
+	actualDigest, err := d.readBytes(uint32(d.h.Size())) //nolint:all
```
Same here about nolint

pavlo-v-chernykh commented on keystore_test.go:
```diff
@@ -246,68 +193,48 @@ func TestLoadKeyPassword(t *testing.T) {
-	if !reflect.DeepEqual(actualPKE.PrivateKey, decodedPK.Bytes) {
-		t.Errorf("unexpected private key %v \n %v", actualPKE.PrivateKey, decodedPK.Bytes)
-	}
+	assert.Truef(t, reflect.DeepEqual(actualPKE.PrivateKey, decodedPK.Bytes),
```
Please consider using `assert.Equal` and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on keystore_test.go:
```diff
@@ -182,60 +146,43 @@ func TestAliases(t *testing.T) {
-	if !reflect.DeepEqual(actualPKE.PrivateKey, decodedPK.Bytes) {
-		t.Errorf("unexpected private key")
-	}
+	assert.True(t, reflect.DeepEqual(actualPKE.PrivateKey, decodedPK.Bytes), "unexpected private key")
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on keystore_test.go:
```diff
@@ -182,60 +146,43 @@ func TestAliases(t *testing.T) {
-	if !reflect.DeepEqual(expectedAliases, actualAliases) {
-		t.Fatal("aliases must be equal")
-	}
+	assert.True(t, reflect.DeepEqual(expectedAliases, actualAliases), "aliases must be equal")
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on keystore_test.go:
```diff
@@ -38,48 +40,31 @@ func TestSetGetMethods(t *testing.T) {
-	if !reflect.DeepEqual(pke.CertificateChain, chainGet) {
-		t.Fatal("certificate chains of private key entries are not equal")
-	}
+	assert.True(t, reflect.DeepEqual(pke, pkeGet), "private key entries not equal")
+	assert.True(t, reflect.DeepEqual(pke.CertificateChain, chainGet),
+		"certificate chains of private key entries are not equal")
+	assert.True(t, reflect.DeepEqual(tce, tceGet), "private key entries not equal")
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -468,17 +439,10 @@ func TestReadCertificate(t *testing.T) {
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
+		assert.Truef(t, reflect.DeepEqual(cert, tt.cert), "invalid certificate '%v' '%v'", cert, tt.cert)
 
 		hash := d.h.Sum(nil)
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -362,18 +340,11 @@ func TestReadString(t *testing.T) {
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
+		assert.Equal(t, tt.string, str)
 
 		hash := d.h.Sum(nil)
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -292,18 +276,12 @@ func TestReadBytes(t *testing.T) {
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -292,18 +276,12 @@ func TestReadBytes(t *testing.T) {
-		if !reflect.DeepEqual(bts, tt.bytes) {
-			t.Errorf("invalid bytes '%v' '%v'", bts, tt.bytes)
-		}
+		assert.Truef(t, reflect.DeepEqual(bts, tt.bytes), "invalid bytes '%v' '%v'", bts, tt.bytes)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -221,20 +212,14 @@ func TestReadUint64(t *testing.T) {
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -221,20 +212,14 @@ func TestReadUint64(t *testing.T) {
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -143,20 +140,14 @@ func TestReadUint32(t *testing.T) {
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -143,20 +140,14 @@ func TestReadUint32(t *testing.T) {
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -69,20 +72,14 @@ func TestReadUint16(t *testing.T) {
-		if !reflect.DeepEqual(hash, tt.hash[:]) {
-			t.Errorf("invalid hash '%v' '%v'", hash, tt.hash)
-		}
+		assert.Truef(t, reflect.DeepEqual(hash, tt.hash[:]), "invalid hash '%v' '%v'", hash, tt.hash)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh commented on decoder_test.go:
```diff
@@ -69,20 +72,14 @@ func TestReadUint16(t *testing.T) {
-		if !reflect.DeepEqual(err, tt.err) {
-			t.Errorf("invalid error '%v' '%v'", err, tt.err)
-		}
+		assert.Truef(t, reflect.DeepEqual(err, tt.err), "invalid error '%v' '%v'", err, tt.err)
```
Please consider using assert.Equal and put expected and actual arguments in a correct order.

pavlo-v-chernykh reviewed (changes requested):
Thank you for your contribution @nitram509 

I left several comments, please address them.

Also, please fix the next linter warnings as well:
```
WARN [config_reader] The configuration option `linters.gomnd.settings` is deprecated. Please use the options `linters.gomnd.checks`,`linters.gomnd.ignored-numbers`,`linters.gomnd.ignored-files`,`linters.gomnd.ignored-functions`.
WARN [lintersdb] The linter "exhaustivestruct" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "scopelint" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "golint" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "interfacer" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "maligned" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The name "goerr113" is deprecated. The linter has been renamed to: err113.
WARN The linter 'execinquery' is deprecated (since v1.58.0) due to: The repository of the linter has been archived by the owner.
WARN The linter 'exportloopref' is deprecated (since v1.60.2) due to: Since Go1.22 (loopvar) this linter is no longer relevant. Replaced by copyloopvar.
WARN The linter 'gomnd' is deprecated (since v1.58.0) due to: The linter has been renamed. Replaced by mnd.
```

nitram509 commented:
Thanks for your comprehensive review.

---

> Also, please fix the next linter warnings as well:
> WARN [config_reader] The configuration option `linters.gomnd.settings` is deprecated. Please use the options `linters.gomnd.checks`,`linters.gomnd.ignored-numbers`,`linters.gomnd.ignored-files`,`linters.gomnd.ignored-functions`.

Just FYI: tried already, that lead to other "warning", which felt like an odd chain.
Will try again.

nitram509 commented on common_test.go:
```diff
@@ -50,13 +53,11 @@ func TestPasswordBytes(t *testing.T) {
-		if !reflect.DeepEqual(output, tt.output) {
-			t.Errorf("convert password bytes '%v', '%v'", output, tt.output)
-		}
+		assert.Truef(t, reflect.DeepEqual(output, tt.output), "convert password bytes '%v', '%v'", output, tt.output)
```
did switch to `.Equal(expected, actual)` as the documentation says.

nitram509 committed:
incorporate suggestions and feedback

nitram509 committed:
fix warning 'The configuration option linters.gomnd.settings is deprecated'
```
WARN [config_reader] The configuration option `linters.gomnd.settings` is deprecated. Please use the options `linters.gomnd.checks`,`linters.gomnd.ignored-numbers`,`linters.gomnd.ignored-files`,`linters.gomnd.ignored-functions`.
```

nitram509 committed:
fix warning 'The linter "$name" is deprecated (step 2) and deactivated'
```
WARN [lintersdb] The linter "exhaustivestruct" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "scopelint" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "golint" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "interfacer" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
WARN [lintersdb] The linter "maligned" is deprecated (step 2) and deactivated. It should be removed from the list of disabled linters. https://golangci-lint.run/product/roadmap/#linter-deprecation-cycle
```

nitram509 committed:
fix warning 'The name "goerr113" is deprecated'
```
WARN [lintersdb] The name "goerr113" is deprecated. The linter has been renamed to: err113.
```

nitram509 committed:
fix warning 'WARN The linter '$name' is deprecated'
```
WARN The linter 'gomnd' is deprecated (since v1.58.0) due to: The linter has been renamed. Replaced by mnd.
WARN The linter 'execinquery' is deprecated (since v1.58.0) due to: The repository of the linter has been archived by the owner.
WARN The linter 'exportloopref' is deprecated (since v1.60.2) due to: Since Go1.22 (loopvar) this linter is no longer relevant. Replaced by copyloopvar.
```

nitram509 commented:
@pavlo-v-chernykh 
I did implement all your suggestions and got rid of the warnings.

pavlo-v-chernykh commented:
Great work on this PR, @nitram509 🎉

Merging now! 🚀

Thanks for your contribution!
